languages/js/src/Oso.ts

```diff
@@ -17,13 +33,202 @@ export class Oso extends Polar {
    * @returns An access control decision.
    */
   async isAllowed(
-    actor: unknown,
-    action: unknown,
-    resource: unknown
+    actor: Actor,
+    action: Action,
+    resource: Resource
   ): Promise<boolean> {
     const results = this.queryRule('allow', actor, action, resource);
     const { done } = await results.next();
     results.return();
     return !done;
   }
+
+  /**
+   * Ensure that `actor` is allowed to perform `action` on
+   * `resource`.
+   *
+   * If the action is permitted with an `allow` rule in the policy, then
+   * this method returns `None`. If the action is not permitted by the
+   * policy, this method will raise an error.
+   *
+   * The error raised by this method depends on whether the actor can perform
+   * the `"read"` action on the resource. If they cannot read the resource,
+   * then a `NotFound` error is raised. Otherwise, a `ForbiddenError` is
+   * raised.
+   *
+   * @param actor The actor performing the request.
+   * @param action The action the actor is attempting to perform.
+   * @param resource The resource being accessed.
+   * @param checkRead If set to `false`, a `ForbiddenError` is always
+   *   thrown on authorization failures, regardless of whether the actor can
+   *   read the resource. Default is `true`.
+   */
+  async authorize(
+    actor: Actor,
+    action: Action,
+    resource: Resource,
+    options: { checkRead?: boolean } = {}
+  ): Promise<void> {
+    if (typeof options.checkRead === 'undefined') options.checkRead = true;
+    if (await this.queryRuleOnce('allow', actor, action, resource)) {
+      return;
+    }
+
+    let isNotFound = false;
+    if (action === this.#readAction) {
+      isNotFound = true;
+    } else if (options.checkRead) {
+      const canRead = await this.queryRuleOnce(
+        'allow',
+        actor,
+        this.#readAction,
+        resource
+      );
```

Comment:
Worth calling `this.isAllowed()` in both of these places?

Reply:
I don't think it's worth it -- I'd prefer to keep the public APIs independent from each other, especially because it doesn't save any lines of code.